app/uk/gov/hmrc/nrs/retrieval/config/HttpHead.scala

```diff
@@ -42,46 +43,47 @@ object CoreHttpReads extends HttpErrorFunctions {
 
   def responseHandler(method: String, url: String, response: HttpResponse): HttpResponse = {
     response.status match {
-      case status if (status == 404) => {
+      case status if status == NOT_FOUND =>
         logger.info(s"Submission bundle not found $status for query $method $url")
         if(method == HEAD_VERB) {
           response
         } else {
-          HttpResponse(404, JsArray.empty, Map[String,Seq[String]]())
+          HttpResponse(NOT_FOUND, JsArray.empty, Map[String,Seq[String]]())
         }
+      case _ => handleResponseEither(method, url)(response) match {
+        case Right(response) => response
+        case Left(err) => throw err
```

Comment:
if err is thrown, what will end user see?

Reply:
err is either an instance of `UpstreamErrorResponse`

https://github.com/hmrc/http-verbs/blob/9ce93fbad29cc298ada2ed8d95eac532664d26d0/http-verbs-common/src/main/scala/uk/gov/hmrc/http/HttpExceptions.scala#L127

so either a `Upstream4xxResponse` or a `Upstream5xxResponse` 

so as per the it:spec the user will see a INTERNAL_SERVER_ERROR or a BAD_GATEWAY post play 2.7 upgrade except for `404` which is preserved.

prior to the upgrade the `400` BAD_REQUEST was also preserved.